com.unity.shadergraph/Editor/Drawing/Inspector/ShaderInputPropertyDrawer.cs

```diff
@@ -0,0 +1,912 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using UnityEditor;
+using UnityEditor.Graphing;
+using UnityEditor.Graphing.Util;
+using UnityEditor.ShaderGraph;
+using UnityEditor.ShaderGraph.Drawing;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEditor.UIElements;
+using UnityEditorInternal;
+using UnityEngine;
+using UnityEngine.UIElements;
+using FloatField = UnityEditor.ShaderGraph.Drawing.FloatField;
+
+namespace Drawing.Inspector
+{
+   // #TODO - This does a lot of things a controller would, think about splitting stuff up to be more MVC
```

Comment:
Remove TODO; either fix it, file a task in Favro to change it, or live with it.

Reply:
Removed all TODOs from code